tools/jit/gen_jit_dispatch.py

```diff
@@ -274,6 +274,32 @@ def emit_decl(decl):
         for variant in {all_real_arguments_are_inputs, only_tensors_are_inputs}:
             emit_decl_variant(decl, variant, has_tensorlist)
 
+    # This function declares an order on declarations. This is necessary because
```

Comment:
This should become unnecessary when we remove the implicit conversion from Tensor -> Number, right? We should make a note of it here, because it is quite complicated.

Reply:
Most likely yes. There's an edge case that I'm worried about (pytorch allows one to pass in a scalar tensor as both a Scalar and Tensor argument, and if we allow that too in jit script the ordering will matter), but I think Sam is working on getting rid of overloads like this. I'll make a note here.